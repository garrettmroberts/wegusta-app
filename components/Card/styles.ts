import { StyleSheet } from 'react-native'
import Colors from '../../constants/Colors'
import Sizes from '../../constants/Sizes'

const styles = StyleSheet.create({
  cardContainer: {
    backgroundColor: Colors.white,
    height: Sizes.cardHeight,
    borderRadius: 20,
    overflow: 'hidden'
  },
  card: {
    height: Sizes.cardHeight,
    width: Sizes.screenWidth - Sizes.viewPadding * 2,
    justifyContent: 'center'
  },
  image: {
    flex: 1,
    resizeMode: 'cover',
    overflow: 'hidden'
  }
})

export default styles
