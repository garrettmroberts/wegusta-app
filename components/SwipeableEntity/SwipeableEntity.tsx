import { ReactNode, useState } from 'react';
import { useRef } from 'react';
import { Animated, View, PanResponder } from 'react-native';

import Sizes from '../../constants/Sizes';

type Props = {
  children?: ReactNode;
  onSwipeLeft?: () => void;
  onSwipeRight?: () => void;
};

const SwipeableEntity = ({ children, onSwipeLeft, onSwipeRight }: Props) => {
  const [visible, setVisible] = useState(true);
  const pan = useRef(new Animated.ValueXY()).current;

  const panResponder = useRef(
    PanResponder.create({
      onMoveShouldSetPanResponder: () => true,
      onPanResponderGrant: () => {
        pan.setOffset({
          //@ts-ignore
          x: pan.x._value,
          //@ts-ignore
          y: pan.y._value
        });
      },
      onPanResponderMove: Animated.event([null, { dx: pan.x, dy: pan.y }], {
        useNativeDriver: false
      }),
      onPanResponderRelease: (evt, gestureState) => {
        if (gestureState.dx < -120) {
          Animated.spring(pan, {
            toValue: {
              x: -Sizes.screenWidth - 100,
              y: gestureState.dy
            },
            useNativeDriver: false
          }).start(() => {
            pan.setValue({ x: 0, y: 0 });
          });
          if (onSwipeLeft) onSwipeLeft();
          setTimeout(() => setVisible(false), 200);
        } else if (gestureState.dx > 120) {
          Animated.spring(pan, {
            toValue: {
              x: Sizes.screenWidth + 100,
              y: gestureState.dy
            },
            useNativeDriver: false
          }).start(() => {
            pan.setValue({ x: 0, y: 0 });
          });
          if (onSwipeRight) onSwipeRight();
          setTimeout(() => setVisible(false), 200);
        } else {
          Animated.spring(pan, {
            toValue: { x: 0, y: 0 },
            friction: 4,
            useNativeDriver: false
          }).start();
        }
      }
    })
  ).current;

  return visible ? (
    <Animated.View
      style={{
        transform: [{ translateX: pan.x }, { translateY: pan.y }]
      }}
      {...panResponder.panHandlers}
    >
      {children}
    </Animated.View>
  ) : (
    <View />
  );
};

export default SwipeableEntity;
