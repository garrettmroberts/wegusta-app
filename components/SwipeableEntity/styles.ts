import { StyleSheet } from 'react-native';
import Colors from '../../constants/Colors';
import Fonts from '../../constants/Fonts';
import Sizes from '../../constants/Sizes';

export default StyleSheet.create({
  container: {
    width: Sizes.screenWidth - Sizes.viewPadding * 2
  },
  text: {
    ...Fonts.displayBold,
    position: 'absolute',
    top: 50,
    borderWidth: 2,
    paddingHorizontal: 10
  },
  likeText: {
    color: Colors.success,
    right: 50,
    transform: [{rotate: '25deg'}],
    backgroundColor: `${Colors.success}25`,
    borderColor: Colors.success
  },
  dislikeText: {
    color: Colors.error,
    left: 40,
    transform: [{rotate: '-25deg'}],
    backgroundColor: `${Colors.error}25`,
    borderColor: Colors.error
  },
  icon: {
    position: 'absolute',
    top: 50
  },
  likeIcon: {
    right: 50,
    transform: [{rotate: '25deg'}],
  },
  dislikeIcon: {
    left: 50,
    transform: [{rotate: '-25deg'}]
  },
});
