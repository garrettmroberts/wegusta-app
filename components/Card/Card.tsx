import { Image, View } from 'react-native';

import styles from './styles';

const Card = () => {
  return (
    <View style={styles.card}>
      <Image
        style={styles.image}
        source={{ uri: 'https://picsum.photos/200/300' }}
      />
    </View>
  );
};

export default Card;
