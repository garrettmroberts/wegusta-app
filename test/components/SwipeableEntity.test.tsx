import { render, screen, fireEvent } from '@testing-library/react-native';

import SwipeableEntity from '../../components/SwipeableEntity/SwipeableEntity';
import Card from '../../components/Card/Card';

describe('<SwipeableEntity />', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <SwipeableEntity>
        <Card imageProps={{ uri: 'https://picsum.photos/200/300' }} />
      </SwipeableEntity>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
