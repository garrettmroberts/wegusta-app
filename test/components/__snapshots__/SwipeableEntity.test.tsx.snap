// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SwipeableEntity /> renders correctly 1`] = `
<View
  collapsable={false}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "transform": Array [
        Object {
          "translateX": 0,
        },
        Object {
          "translateY": 0,
        },
      ],
    }
  }
>
  <View
    style={
      Object {
        "height": 1034,
        "padding": 10,
        "position": "absolute",
        "width": 750,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "borderRadius": 20,
          "flex": 1,
          "overflow": "hidden",
          "resizeMode": "cover",
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": "https://picsum.photos/200/300",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
